api/v1/server/handlers/monitoring/probe.go

```diff
@@ -0,0 +1,317 @@
+package monitoring
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/labstack/echo/v4"
+	"golang.org/x/exp/rand"
+
+	"github.com/hatchet-dev/hatchet/api/v1/server/oas/gen"
+	"github.com/hatchet-dev/hatchet/pkg/client"
+	"github.com/hatchet-dev/hatchet/pkg/cmdutils"
+	clientconfig "github.com/hatchet-dev/hatchet/pkg/config/client"
+	"github.com/hatchet-dev/hatchet/pkg/config/shared"
+	"github.com/hatchet-dev/hatchet/pkg/repository/prisma/db"
+	"github.com/hatchet-dev/hatchet/pkg/repository/prisma/sqlchelpers"
+	"github.com/hatchet-dev/hatchet/pkg/worker"
+)
+
+func (m *MonitoringService) MonitoringPostRunProbe(ctx echo.Context, request gen.MonitoringPostRunProbeRequestObject) (gen.MonitoringPostRunProbeResponseObject, error) {
+	if !m.enabled {
+		return nil, fmt.Errorf("monitoring is not enabled")
+	}
+
+	tenant := ctx.Get("tenant").(*db.TenantModel)
+
+	if tenant == nil || tenant.ID == "" {
```

Comment:
This check already exists in the `authn` middleware, which checks that the authorization token belongs to the tenant which is set in the context.